### PR TITLE
updates test and refactors component for RevisionsOverviewList

### PR DESCRIPTION
**Pull request by invincibleJai**
updates test and refactors component for RevisionsOverviewList

- Refactored `RevisionsOverviewList` into two component `RevisionsOverviewListItem` and `RevisionsOverviewList`
- Added unit tests for `RevisionsOverviewList`
- Added unit tests for `RevisionsOverviewListItem`

Tracks: https://jira.coreos.com/browse/ODC-2513

Screenshot

Screenshot 2019-12-11 at 5 42 21 PM

### CODE REVIEW

File: frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewListItem.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.odc-revision-deployment-list {
+    padding: 15px 15px 0px 20px;
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewListItem.scss]:
```suggestion
    padding: 15px 20px 0;
```
Padding right and left: 20px, top: 15px; and bottom: 0;

File: frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewList.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    expect(wrapper.find(Button).exists()).toBe(false);
+  });
+
+  it('should render RevisionsOverviewListItem', () => {
+    const wrapper = shallow(
+      <RevisionsOverviewList revisions={mockRevisionsData} service={mockServiceData} />,
+    );
+    expect(wrapper.find(RevisionsOverviewListItem)).toHaveLength(1);
+  });
+});
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewList.spec.tsx]: Can you also add a test case for ```onClick``` on ```Set Traffic Distribution``` button?

Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewList.spec.tsx]: Added

File: frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewListItem.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.odc-revision-deployment-list {
+    padding: 15px 20px 0;
+    &__pod {
+      width: 25px;
+      cursor: default;
+    }
```
Comment (divyanshiGupta) [inline frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewListItem.scss]: Can we use pf variables for padding and width here?

File: frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewListItem.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    expect(
+      wrapper
+        .find('span')
+        .at(0)
+        .props().children,
+    ).toEqual('100%');
+  });
+
+  it('should not show deployments if not present', () => {
+    expect(wrapper.find('.odc-revision-deployment-list').exists()).toBe(false);
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewListItem.spec.tsx]:
```suggestion
    expect(wrapper.find('.odc-revision-deployment-list').exists()).toBeFalsy();
```

File: frontend/packages/knative-plugin/src/components/overview/__tests__/RoutesOverviewList.spec.tsx [hunk 5]
```diff
@@ -27,26 +26,12 @@ describe('RoutesOverviewList', () => {
...
-        .at(0)
-        .props().className,
-    ).toEqual('text-muted');
-    expect(
-      wrapper
-        .find('span')
-        .at(0)
-        .props().children,
-    ).toEqual('No Routes found for this resource.');
+    expect(wrapper.text().includes('No Routes found for this resource.')).toBe(true);
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/__tests__/RoutesOverviewList.spec.tsx]:
```suggestion
    expect(wrapper.text().includes('No Routes found for this resource.')).toBeTruthy();
```

File: frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewListItem.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  });
+
+  describe('should show deployments if present', () => {
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewListItem.spec.tsx]: ```describe``` should be the name of the test suit.

Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewListItem.spec.tsx]: @vikram-raj  updated, ptal

**Comment by spadgett** [comment]
/hold
e2e-gcp-console is currently broken

**Comment by spadgett** [comment]
/hold cancel
/retest
#3847 has fixed the e2e tests

**Pull request merged**